numpy/fft/tests/test_pocketfft.py

```diff
@@ -465,6 +465,28 @@ def test_irfftn_out_and_s_interaction(self, s):
         assert result is out
         assert_array_equal(result, expected)
 
+    def test_fftn_with_out_and_s(self):
+        # fftn works correctly with both out and s parameters
+        rng = np.random.RandomState(42)
+        x = np.random.rand(4, 4, 4) + 1j * np.random.rand(4, 4, 4)
```

Comment:
It may be wise to ensure we never modify the input, by doing after the definition, `x.flags.writeable = False`.